services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -987,6 +987,118 @@ async def _process_s3_page_results(
 
         return current_page_results
 
+    def _process_s3_directory_entry(
+        self,
+        user_id: UserID,
+        s3_entry: S3DirectoryMetaData,
+        name_pattern_lower: str,
+        min_parts_for_valid_s3_object: int,
+    ) -> FileMetaData | None:
+        prefix_str = str(s3_entry.prefix)
+
+        # Handle both cases: with and without trailing slash
+        # For prefixes like "project_id/node_id/directory_name/" or "project_id/node_id/directory_name"
+        normalized_prefix = prefix_str.rstrip("/")
+        prefix_parts = normalized_prefix.split("/")
```

Comment:
This is equivalent and more robust
```suggestion
        prefix_parts = Path(prefix_str).parts
```